From 499426e4f54aee9169494918c2d3c12d56099ab9 Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Fri, 27 Jan 2023 17:05:44 -0500
Subject: [PATCH 30/86] Revert "Bug#34778746 use-after-free after
 async_send_tls() in mock-server"

This reverts commit 83701ef1f3464a06e1fcabcd4ff1029767ddb9fd.
---
 router/src/mock_server/src/statement_reader.h | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/router/src/mock_server/src/statement_reader.h b/router/src/mock_server/src/statement_reader.h
index 5ee51af8e1c..235b2ab3129 100644
--- a/router/src/mock_server/src/statement_reader.h
+++ b/router/src/mock_server/src/statement_reader.h
@@ -147,18 +147,16 @@ class ProtocolBase {
               async_send(std::move(compl_handler));
             });
       } else {
-        net::defer(client_socket_.get_executor(),
-                   [compl_handler = std::move(init.completion_handler),
+        net::defer([compl_handler = std::move(init.completion_handler),
                     ec = write_res.error()]() { compl_handler(ec, {}); });
       }
     } else {
       net::dynamic_buffer(send_buffer_).consume(write_res.value());
 
-      net::defer(client_socket_.get_executor(),
-                 [compl_handler = std::move(init.completion_handler),
+      net::defer([compl_handler = std::move(init.completion_handler),
                   transferred = write_res.value()]() {
-                   compl_handler({}, transferred);
-                 });
+        compl_handler({}, transferred);
+      });
     }
 
     return init.result.get();
-- 
2.37.2

